src/connector/postgres/error.rs

```diff
@@ -335,7 +335,19 @@ fn try_extracting_io_error(err: &tokio_postgres::error::Error) -> Option<Error>
 
     err.source()
         .and_then(|err| err.downcast_ref::<std::io::Error>())
-        .map(|err| ErrorKind::ConnectionError(Box::new(std::io::Error::new(err.kind(), format!("{}", err)))))
+        .map(|err| {
+            let inner_io_error = Box::new(std::io::Error::new(err.kind(), format!("{}", err)));
+            match err.kind() {
+                // tokio_postgres will throw `std::io::Error`s with kind `InvalidInput` in several occasions,
+                // e.g., when the number of provided bind variables is >= 32768.
+                // We map these to `quaint::error::ErrorKind::QueryInvalidInput` to avoid confusing user-facing
+                // error messages.
+                std::io::ErrorKind::InvalidInput => ErrorKind::QueryInvalidInput(inner_io_error),
+
+                // fall back to a generic `ConnectionError`
+                _ => ErrorKind::ConnectionError(inner_io_error),
```

Comment:
Does this make sense as the generic fallback to `std::io::Error`? Isn't the danger that this covers still to many other cases, where we will keep having a unexpected "Can't connect ..." error message instead of anything helpful? Is there maybe a way to get at least the error message and include it as meta information?

Reply:
The inner message is already included as "meta" information in `inner_io_error`. Currently, it gets lost elsewhere in the `query-engine`. I'll gladly fix that once this and [the companion PR](https://github.com/prisma/prisma-engines/pull/3201) are merged.

Reply:
That would be amazing, as it would make bugs like the one you are fixing actually potentially debuggable.